statistics_calculations.py

```diff
@@ -4,30 +4,29 @@
 from parsing_tools import Message
 import matplotlib.pyplot as plt
 
-def count_word(df, word):
-    """"""  # TODO - add docstring and type hints
+def count_word(df: pd.DataFrame, word: str) -> pd.DataFrame:
     """ Count use of word per author. The word is passed in the form of regex."""
     return df[df["text"].str.contains(r"{}($|\s)".format(word))].groupby(
         "author").count()
 
 
-def count_haha(df):
-    """"""  # TODO - add docstring and type hints
+def count_haha(df: pd.DataFrame) -> pd.DataFrame:
+    """Count how many laugh messages were sent by authors"""
     return df[df["text"].str.contains(r"ח{3,}")].groupby("author").count()
 
 
-def count_emoji(df):
-    """"""  # TODO - add docstring and type hints
+def count_emoji(df: pd.DataFrame) -> pd.DataFrame:
+    """Count how many emojis were used by each author"""

```

Comment:
Actually it seems like we count messages with emojis, don't we?
If I use two emojis in the same message, that will count as one?
I think it happens in all of our counter functions (messages with X instead of number of X)
Should I add an issue to change that? or should we change the docu? I believe number of emojis is more interesting than number of messages with emojis.

Reply:
Yeah, I agree - what we want it to count number of uses (emojis, questions, curses...) and not number of messages with them. We should open an issue to modify the functions I think

Reply:
Opened